ui/src/app/applications/components/utils.tsx

```diff
@@ -497,32 +497,59 @@ export const deletePopup = async (
 };
 
 export function getResourceActionsMenuItems(resource: ResourceTreeNode, metadata: models.ObjectMeta, apis: ContextApis): Promise<ActionMenuItem[]> {
-    return services.applications
-        .getResourceActions(metadata.name, metadata.namespace, resource)
-        .then(actions => {
-            return actions.map(
-                action =>
-                    ({
-                        title: action.displayName ?? action.name,
-                        disabled: !!action.disabled,
-                        iconClassName: action.iconClass,
-                        action: async () => {
+    return services.applications.getResourceActions(metadata.name, metadata.namespace, resource).then(actions => {
+        return actions.map(action => ({
+            title: action.displayName ?? action.name,
+            disabled: !!action.disabled,
+            iconClassName: action.iconClass,
+            action: async () => {
+                const confirmed = false;
+                const title = action.hasParameters ? `Enter input parameters for action: ${action.name}` : `Execute ${action.name} action?`;
+                console.log('title:', title);
+                await apis.popup.prompt(
+                    title,
+                    api => (
+                        <div>
+                            {!action.hasParameters && (
+                                <div className='argo-form-row'>
+                                    <div> Are you sure you want to execute {action.name} action?</div>
```

Comment:
I'm not a huge fan of `execute` in user facing content. 


```suggestion
                                    <div> Are you sure you want to perform {action.name} action?</div>
```